src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java

```diff
@@ -171,6 +180,26 @@ public String getPartitionTo() {
     return Strings.isNullOrEmpty(partitionTo) ? null : partitionTo;
   }
 
+  @Nullable
+  public String getFilter() {
+    if (filter != null) {
+      filter = filter.trim();
+      if (filter.isEmpty()) {
+        return null;
+      }
+      // remove the WHERE from the filter
+      if (filter.toUpperCase().startsWith(WHERE)) {
+        filter = filter.replace(WHERE, "");
```

Comment:
this doesn't work if the 'where' is not all uppercase. It's better to use substring()

Reply:
changing this to:

`filter = filter.substring(WHERE.length());`